add project and open project dialogs

alanrenmsft opened the pull request:
1. adjust the view toolbar as we discusssed in the sync up meeting.
primary actions: New Project, Open Project
secondary actions: Refresh
![image](https://user-images.githubusercontent.com/13777222/94999480-7efd4100-056e-11eb-9fc8-6ca527efd237.png)

![image](https://user-images.githubusercontent.com/13777222/94999494-93413e00-056e-11eb-93c6-c18dfbd58a13.png)

2. implement new project and open project dialogs
the view will be refreshed when project is opened or added.

open project dialog
![image](https://user-images.githubusercontent.com/13777222/94999501-9b00e280-056e-11eb-94d8-0d0cd7da4eed.png)

file picker
![image](https://user-images.githubusercontent.com/13777222/94999524-c388dc80-056e-11eb-9695-ada1c62a038d.png)

open project dialog error case
![image](https://user-images.githubusercontent.com/13777222/94999514-b66bed80-056e-11eb-869f-7d759473d36b.png)

add project dialog
![image](https://user-images.githubusercontent.com/13777222/94999535-d69bac80-056e-11eb-832b-c228f083bae1.png)

folder picker
![image](https://user-images.githubusercontent.com/13777222/94999613-5de92000-056f-11eb-91b3-4abaf1e866d7.png)

add project error - folder not exist
![image](https://user-images.githubusercontent.com/13777222/94999634-8a04a100-056f-11eb-8806-a2c61955dd89.png)

add project error - folder with same name already exist
![image](https://user-images.githubusercontent.com/13777222/94999640-a1438e80-056f-11eb-9921-ad9286b0c2f8.png)

3. welcome view update
no workspace
![image](https://user-images.githubusercontent.com/13777222/95003382-9995e100-0593-11eb-9195-6804b1937993.png)


workspace and no project
![image](https://user-images.githubusercontent.com/13777222/95003365-808d3000-0593-11eb-95ea-43e5ca04742d.png)

kisantia commented on extensions/data-workspace/src/dialogs/newProjectDialog.ts:
```diff
@@ -0,0 +1,151 @@
+			this.model.location = locationTextBox.value!;
+		}));
+
+		const browseFolderButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({ label: constants.BrowseButtonText, width: constants.DefaultButtonWidth }).component();
```
browse buttons should be folder icons to match the mockup

alanrenmsft replied:
I'll consistent with other places in ADS

kisantia commented on extensions/data-workspace/package.nls.json:
```diff
@@ -3,8 +3,10 @@
 	"remove-project-command": "Remove Project",
-	"projects-view-no-workspace-content": "To use projects, open a workspace and add projects to it, or use the 'Add Project' feature and we will create a workspace for you.\n[Open Workspace](command:workbench.action.openWorkspace)\n[Add Project](command:projects.addProject)"
+	"projects-view-no-workspace-content": "You need to open a workspace in order to use the project feature.\n[Open Workspace…](command:workbench.action.openWorkspace)\n[Save Workspace…](command:workbench.action.saveWorkspaceAs)\n",
+	"projects-view-no-project-content": "No project found in current workspace.\n[New Project…](command:projects.newProject)\n[Open Project…](command:projects.openProject)\n",
```
"No projects" instead of "No project"

kisantia commented on extensions/data-workspace/src/common/utils.ts:
```diff
@@ -0,0 +1,31 @@
+}
+
+export async function fileExist(filePath: string): Promise<boolean> {
+	const stats = await getFileStatus(filePath);
```
a lot of extensions have an exists function like this: https://github.com/microsoft/azuredatastudio/blob/ef0a92d83ffe8b169f50aba1e9ef7f77eb94491a/extensions/agent/src/agentUtils.ts#L52-L53

Is there a difference between using that and the one here?

alanrenmsft replied:
yes, we don't have a good way to share code between extensions. for this specific method, my requirement is more strict, I need to make sure it is a directory or file.

kisantia commented on extensions/sql-database-projects/src/common/constants.ts:
```diff
@@ -23,7 +23,9 @@ export const MicrosoftDatatoolsSchemaSqlSql = 'Microsoft.Data.Tools.Schema.Sql.S
 // Project Provider
-export const projectTypeDisplayName = localize('projectTypeDisplayName', 'Database Project');
+export const sqlDatabaseProjectTypeId = 'sqldbproj';
+export const projectTypeDisplayName = localize('projectTypeDisplayName', "Database Project");
```
Should this be SQL Database Project to clarify?

alanrenmsft replied:
more information is available in description

kisantia commented on extensions/data-workspace/src/dialogs/newProjectDialog.ts:
```diff
@@ -0,0 +1,151 @@
+			this.model.projectTypeId = e.cardId;
+		}));
+
+		const projectNameTextBox = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
```
should there be an info icon that says that a new folder will get created in the selected directory with the project name?

alanrenmsft replied:
this is the same experience as the current sql-db-proj extension.

kisantia commented on extensions/sql-database-projects/src/projectProvider/projectProvider.ts:
```diff
@@ -39,9 +44,22 @@ export class SqlDatabaseProjectProvider implements dataworkspace.IProjectProvide
+	 * @returns Uri of the newly created project file
+	 */
+	async createProject(name: string, location: vscode.Uri): Promise<vscode.Uri> {
+		const projectFile = await this.projectController.createNewProject(name, location, true);
```
the MainController.createNewProject() should be removed since it isn't being used anymore, but I can clean it up later

alanrenmsft commented:
@kisantia I added this and modified the sql-database-project code to handle the extension activation.
![image](https://user-images.githubusercontent.com/13777222/95272067-48d0f300-07f4-11eb-858b-5956ba3c6262.png)
